trieste/acquisition/function.py

```diff
@@ -485,6 +498,10 @@ def __init__(self, sample_size: int):
         tf.debugging.assert_positive(sample_size)
         self._sample_size = sample_size
 
+    def __repr__(self) -> str:
+        """"""
+        return f"SingleModelMCIndAcquisitionFunctionBuilder({self._sample_size!r})"
```

Comment:
Maybe use `{type(self).__name__}` here, so that this is robust to any changes to the class name?

Reply:
i used to do that until someone suggested i didn't. I'll leave it as is for now, because I've done it like this across the code base so it's consistent. I don't really have a preference in general

Reply:
i've actually just noticed these abstract classes shouldn't have `__repr__`s anyway, so I'll delete them